lib/auth/auth_with_roles_test.go

```diff
@@ -4633,3 +4633,239 @@ func TestGenerateCertAuthorityCRL(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateSnowflakeSession(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
```

Comment:
nit: Any reason why we need this context here? It's not being canceled anywhere and moving closer to where it is used/reducing scope IMO would make the test easier to understand.